source/library/Monadoc/Template/Version/Get.hs

```diff
@@ -60,61 +60,55 @@ render context input = do
           ("Package" F.%+ F.stext F.%+ "version" F.%+ F.stext F.%+ ":: Monadoc")
           (Witch.from packageName)
           (Witch.from reversion)
-  Common.base context route (breadcrumbs input) title $ do
-    showDeprecationWarning packageName reversion $ upload input
-    showLatestInfo context packageName (maybeLatest input) $ const Nothing
+  Common.base context route input.breadcrumbs title $ do
+    showDeprecationWarning packageName reversion input.upload
+    showLatestInfo context packageName input.maybeLatest $ const Nothing
     Html.h2_ $ Html.toHtml packageName
     Html.p_ $ do
       "Version "
       Html.toHtml versionNumber
       " revision "
       Html.toHtml revision
       " uploaded "
-      Common.timestamp . Upload.uploadedAt . Model.value $ upload input
+      Common.timestamp input.upload.value.uploadedAt
       " by "
-      Html.a_ [Html.href_ . Common.route context . Route.User . HackageUser.name . Model.value $ hackageUser input]
-        . Html.toHtml
-        . HackageUser.name
-        . Model.value
-        $ hackageUser input
+      Html.a_ [Html.href_ . Common.route context $ Route.User input.hackageUser.value.name] $
+        Html.toHtml input.hackageUser.value.name
       "."
     Html.h3_ "Package meta"
     Html.dl_ $ do
       Html.dt_ "Synopsis"
-      Html.dd_ . maybe "n/a" Html.toHtml . PackageMeta.synopsis . Model.value $ packageMeta input
+      Html.dd_ . maybe "n/a" Html.toHtml $ input.packageMeta.value.synopsis
       Html.dt_ "Description"
       Html.dd_
         . Html.toHtml
         . Haddock.markup (markup context)
         . Haddock.overIdentifier (curry Just)
         . Haddock._doc
         . Haddock.parseParas Nothing
-        . maybe "" (Witch.into @String)
-        . PackageMeta.description
-        . Model.value
-        $ packageMeta input
+        $ maybe "" (Witch.into @String) input.packageMeta.value.description
       Html.dt_ "Author"
-      Html.dd_ . maybe "n/a" Html.toHtml . PackageMeta.author . Model.value $ packageMeta input
+      Html.dd_ . maybe "n/a" Html.toHtml $ input.packageMeta.value.author
```

Comment:
I wish that HLint could flag `f $ x.y` as an unnecessary dollar sign and suggest replacing it with `f x.y`.